lib/active_resource/collection_ext.rb

```diff
@@ -0,0 +1,7 @@
+require 'shopify_api/collection_pagination'
+
+module ActiveResource
+  class Collection
+    prepend ShopifyAPI::CollectionPagination
+  end
+end
```

Comment:
as discussed IRL do we have to solve the problem of querying the next/prev link after doing another query?, e.g.

```
page1 = Produdcts.find(:all)
page2 = Produdcts.find(:all)
page2.fetch_next_page
page1.next_page
page1.previous_page
```

Reply:
Updated to parse the headers on collection initialization, vs lazily evaluating them. Also added a test for that case. The collection objects don't have access to the actual response, so we have to rely on `ShopifyAPI::Base.connection.response["Link"]`, which is always for the last made request.